integreat_cms/static/src/js/pois/poi-map.ts

```diff
@@ -1,165 +1,242 @@
-
-        if (initial_longitude && initial_latitude) {
-            prepareMap(initial_latitude, initial_longitude);
+    // Check if an address could be found
+    if (response.status == 200) {
+        // Wait for the response
+        const data = await response.json();
+        // Hide loading spinner
+        document.getElementById("address_loading").classList.add("hidden");
+        // Show found address
+        const street = constructStreet(data.street, data.number);
+        const updatedStreet = document.getElementById("updated_street");
+        if (street) {
+            updatedStreet.textContent = street;
+            updatedStreet.classList.remove("hidden");
         } else {
-            document.getElementById("map").classList.add("hidden");
-            document.getElementById("no_coordinates_alert")?.classList.remove("hidden");
+            updatedStreet.textContent = "";
+            updatedStreet.classList.add("hidden");
         }
-    });
+        document.getElementById("updated_postcode").textContent = data.postcode;
+        document.getElementById("updated_city").textContent = data.city;
+        document.getElementById("updated_country").textContent = data.country;
+        document.getElementById("update_address_from_map_marker").classList.remove("hidden");
+        // Enable address-related radio buttons
+        Array.from(document.getElementsByClassName("depends-on-address")).forEach((element) => {
+            element.classList.remove("cursor-not-allowed", "text-gray-500");
+            if (element instanceof HTMLInputElement) {
+                element.disabled = false;
+            }
+        });
+    } else {
+        // Hide loading spinner
+        document.getElementById("address_loading").classList.add("hidden");
+        // Hide all address-dependent input fields in case they were shown before
+        document.getElementById("update_address_from_map_marker").classList.add("hidden");
+        // Disable address-related radio buttons
+        Array.from(document.getElementsByClassName("depends-on-address")).forEach((element) => {
+            element.classList.add("cursor-not-allowed", "text-gray-500");
+            if (element instanceof HTMLInputElement) {
+                element.disabled = true;
+                element.checked = false;
+            }
+        });
+        // Show error
+        document.getElementById("no_address_found").classList.remove("hidden");
+    }
 }
 
-function construct_address(number: string, street: string, postcode: string, city: string, country: string) {
-    let address = ", " + postcode + ", " + city + ", " + country;
-
-    if (number.length != 0 && street.length != 0) {
-        address = street + " " + number + address;
-    } else if (street.length != 0) {
-        address = street + address;
-    } else if (number.length != 0) {
-        address = number + address;
```

Comment:
I think a number without street does not make sense... I would only show the number when the street is found as well.